packages/react-components/react-tree/src/hooks/useFlatTreeItems.ts

```diff
@@ -0,0 +1,155 @@
+import { useControllableState, useEventCallback } from '@fluentui/react-utilities';
+import * as React from 'react';
+import { TreeItemId, TreeOpenChangeData, TreeOpenChangeEvent, TreeProps } from '../Tree';
+import { TreeItemProps } from '../TreeItem';
+import { ImmutableSet } from '../utils/ImmutableSet';
+import { updateOpenItems } from '../utils/updateOpenItems';
+
+export type FlatTreeItemProps = Required<Pick<TreeItemProps, 'id'>> &
+  TreeItemProps & {
+    parentId?: string;
+  };
+
+type TreeItemPropsReference = {
+  props: Required<Pick<TreeItemProps, 'id' | 'aria-level' | 'aria-posinset' | 'leaf'>> & TreeItemProps;
+  parentId?: string;
+  childrenSize: number;
+};
+
+type TreeItemPropsReferences = {
+  refs: TreeItemPropsReference[];
+  getParent(ref: TreeItemPropsReference): TreeItemPropsReference;
+};
+
+type LazyArray<Value> = {
+  map<NextValue>(fn: (child: Value) => NextValue): NextValue[];
+  toArray(): Value[];
+};
+
+export type FlatTreeProps = Pick<TreeProps, 'openItems' | 'onOpenChange'>;
+export type LazyFlatTreeItems = LazyArray<
+  Required<Pick<TreeItemProps, 'id' | 'aria-level' | 'aria-posinset' | 'leaf' | 'aria-setsize'>> & TreeItemProps
+>;
+
+export type UseFlatTreeItemsOptions = Pick<TreeProps, 'openItems' | 'defaultOpenItems'>;
+
+export function useFlatTreeItems_unstable(
+  items: FlatTreeItemProps[],
+  options?: UseFlatTreeItemsOptions,
+): readonly [FlatTreeProps, LazyFlatTreeItems] {
+  const [openItems, setOpenItems] = useControllableState({
+    state: React.useMemo(() => options?.openItems && ImmutableSet.from(options.openItems), [options?.openItems]),
+    defaultState: React.useMemo(() => options?.defaultOpenItems && ImmutableSet.from(options?.defaultOpenItems), [
+      options?.defaultOpenItems,
+    ]),
+    initialState: ImmutableSet.emptySet,
+  });
+  const onOpenChange = useEventCallback((ev: TreeOpenChangeEvent, data: TreeOpenChangeData) =>
```

Comment:
This shouldn't be `onOpenChange`. In other components that callback name is used to _**notify**_ when something changes. Here the usage is to actually do the change.